app/model/RuleContext.scala

```diff
@@ -33,11 +32,15 @@ case class RuleContext(authContext: AuthContext)(implicit hc: HeaderCarrier) {
     userDetailsConnector.getUserDetails(authority.userDetailsLink)
   }
 
-  lazy val activeEnrolments = enrolments.map { enrolmentSeq =>
+  lazy val activeEnrolmentKeys = enrolments.map { enrolmentSeq =>
```

Comment:
Can you not write it like 
`lazy val activeEnrolmentKeys = activeEnrolments.map(_.key).toSet[String]`

Is this less readable? And it is going to reuse the existing `lazy val` so filtering will happen only once.